examples/simulation/Tutorial1/macros/run_tutorial1.C

```diff
@@ -98,7 +98,13 @@ void run_tutorial1(Int_t nEvents = 10, TString mcEngine = "TGeant3", Bool_t isMT
   // ------------------------------------------------------------------------
 
   // -----   Initialize simulation run   ------------------------------------
+  UInt_t randomSeed = 123456;
+  TRandom3 random(randomSeed);
+  gRandom = &random;
+
   run->Init();
+
+  TVirtualMC::GetMC()->SetRandom(&random);
```

Comment:
Isn't it enough to use

gRandom->SetSeed(123456) ?

Reply:
The method TVirtualMC::SetRandom stores the pointer in both gRandom and fRandom variables. If the second one is also used by the transport engine, then it is not enough.